soundbay/results_analysis.py

```diff
@@ -0,0 +1,108 @@
+import pandas as pd
+import torch
+from torch.utils.data import DataLoader
+import numpy as np
+from pathlib import Path
+import os
+import pandas
+import datetime
+from scipy.special import softmax
+from soundbay.utils.logging import Logger
+
+
+def analysis_logging(results_df,num_classes):
+    """
+    Logs the results of the analysis to a wandb.
+
+    Args:
+        results_df: The dataframe containing the results of the analysis.
+        gt_metadata: The ground truth metadata.
+
+    Returns:
+        None
+
+    """
+    columns_array = results_df.columns[-int(num_classes):]
+    results_array = np.array(results_df[columns_array])
+    metrics_dict = Logger.get_metrics_dict(results_df["label"].values.tolist(),
+                                       np.argmax(results_array, axis=1).tolist(),
+                                       results_array)
+    print(metrics_dict)
+
+def inference_csv_to_raven(probsdataframe: pd.DataFrame, num_classes, seq_length: float, selected_class: str,threshold: float = 0.5, class_name: str = "call") -> pd.DataFrame:
+    # transforms the probability dataframe to a raven format with class_1 predictions as the annotated bounding boxes
+    relevant_columns = probsdataframe.columns[-int(num_classes):]
+    relevant_columns_df = probsdataframe[relevant_columns]
+    len_dataset = relevant_columns_df.shape[0]  # number of segments in wav
+    seq_len = seq_length
+    if_positive = probsdataframe[selected_class] > threshold  # check if the probability is above the threshold
+    if "begin_time" in probsdataframe.columns: #if the dataframe has metadata
+        all_begin_times = probsdataframe["begin_time"].values
+
+    else: #if the dataframe came from a file with no ground truth
+        all_begin_times = np.arange(0, len_dataset * seq_len, seq_len)
+
+    begin_times = all_begin_times[if_positive]
```

Comment:
How do you know that probsdataframe and all_begin_times have the same shape? For e.g. if all_begin_times  was generated in the `else:` path

Reply:
The `len_dataset` is taken from the length of the inference dataframe. In the `else` case it's just the length of the audio file (divided by seq_length) and in the `if` case it's a list of segments from the test set metadata